### PR TITLE
Text Detection: add ppocr-v2 detect -WIP

### PR DESCRIPTION
**Pull request by the-star-sea**
- [x] add demo
- [x] add example in readme

### CODE REVIEW

File: models/text_detection_ppdetect/ppdetect.py
Comment (zihaomu) [inline models/text_detection_ppdetect/ppdetect.py:5]: Please delete this copyright.

**Review by zihaomu — COMMENTED** [review]
Thanks for your contribution!
Newly added models do not need the old copyright.

**Comment by zihaomu** [comment]
Hi @the-star-sea. Since in OpenCV the `DB` has been supported by High-Level API, can you provide more speed and accuracy test data for DB and proposed ppocr?

**Comment by the-star-sea** [comment]
> Hi @the-star-sea. Since in OpenCV the `DB` has been supported by High-Level API, can you provide more speed and accuracy test data for DB and proposed ppocr?

ok.I will do it

**Comment by zihaomu** [comment]
> 

@the-star-sea the ppocr's speed is super fast. Are the pre-process and the post-process of ppocr's the same as the DB net?

**Comment by the-star-sea** [comment]
> @the-star-sea the ppocr's speed is super fast. two questions:
> 
> 1. Are the pre-process and the post-process of ppocr's the same as the DB net?
> 2. Is there accuracy test result?

1.almost the same
2.

**Comment by zihaomu** [comment]
Thank you @the-star-sea! The accuracy result is good. Which validation data are used? How many data it has?

**Comment by the-star-sea** [comment]
> 

icdar2015.500 imgs.

**Comment by zihaomu** [comment]
Thanks for the clarification. The `DB model` has hidded the post-process in the high level api, some thing like the following:
```py

model = cv.dnn_TextDetectionModel_DB(
            cv.dnn.readNet(self._modelPath)
        )

# time start
model.detect(image)
# time stop
```
In order to get a fair speed comparison result, how did you test the speed of ppocr?

**Comment by the-star-sea** [comment]
> 

I write the config file in benchmark and add function support in ppdetect.It just tests the speed of infering onnx model.
Besides.the onnx filesize of ppdetect is 2284 while DB is 47628.

**Comment by the-star-sea** [comment]
> 

I write the config file in benchmark and add function support in ppdetect.It just tests the speed of infering onnx model.
Besides.the onnx filesize of ppdetect is 2284 while DB is 47628.

**Comment by the-star-sea** [comment]
> 

I write the config file in benchmark and add function support in ppdetect.It just tests the speed of infering onnx model.
Besides.the onnx filesize of ppdetect is 2284 while DB is 47628.

**Comment by zihaomu** [comment]
> 47628

That's a great answer. Thanks! 
Can we re-use the High-Level API of DB for PPocr with no change or little change?
```py
model = cv.dnn_TextDetectionModel_DB(
            cv.dnn.readNet(modelPath_ppocr)
        )
model.detect(image)
```

**Comment by the-star-sea** [comment]
> 

I think there need some changes because ppocr doesnot need polygon threhold.I will do it.

**Comment by zihaomu** [comment]
@the-star-sea Please keep one in the mobile and normal models, as we discussed before, to avoid confusing users.

**Comment by the-star-sea** [comment]
> @the-star-sea Please keep one in the mobile and normal models, as we discussed before, to avoid confusing users.

ok.I will do it

File: models/text_detection_ppdetect/README.md
Comment (zihaomu) [inline models/text_detection_ppdetect/README.md:5]: How about adding the original model link here?

**Review by zihaomu — APPROVED** [review]
LGTM!

File: models/__init__.py
Comment (fengyuentau) [inline models/__init__.py:14]: New line after this line

File: models/__init__.py
Comment (fengyuentau) [inline models/__init__.py:40]: EOL before EOF

**Review by fengyuentau — COMMENTED** [review]
I suggest we remove DB from opencv_zoo after this pull request is merged, make a new high level api specifically for this model in opencv and update the wrapper class.

**Comment by zihaomu** [comment]
> I suggest we remove DB from opencv_zoo after this pull request is merged.

Hi @fengyuentau, how about leaving it to the next PR to remove `DB` model, because we have a lot of speed tests based on the `DB` model. In addition, the high-level API based `ppocr` model has not been completed and is expected to be completed by the end of September.

**Comment by fengyuentau** [comment]
Sure, thats exactly what I meant. Just want to emphasize this plan.

File: models/__init__.py
Comment (zihaomu) [reply models/__init__.py:14]: @the-star-sea, please take a look.

Comment (the-star-sea) [reply models/__init__.py:14]: Ok

**Comment by fengyuentau** [comment]
@zihaomu Please update benchmark results on this model.

**Comment by zihaomu** [comment]
> @zihaomu Please update benchmark results on this model.

Hi, @fengyuentau the benchmark results will be updated at PR #73, the student will complete it in the near future.

**Comment by fengyuentau** [comment]
When we add or update a model in opencv zoo, we always update the average forward latency in the same pull request.

**Comment by zihaomu** [comment]
Ok, let's wait.

**Comment by fengyuentau** [comment]
We dont need to wait for #73 as it is for accuracy. At least we need the speed of this model to merge a pull request of adding or updating a model.

**Comment by zihaomu** [comment]
After [this PR](https://github.com/opencv/opencv/pull/22614) is merged, PP-OCR_v3 can be supported and loaded with high-level API. And I think we can only put PP-OCR_v3 in `opencv_zoo` since it has better accuracy.

File: models/text_detection_ppdetect/demo.py
Comment (fengyuentau) [reply models/text_detection_ppdetect/demo.py:29]: Thanks for the review. IIRC, this link was added before we have the wiki in opencv. Will update this in a separate pull request.

**Comment by fengyuentau** [comment]
@zihaomu Anything else is blocking this PR from merge?

**Comment by zihaomu** [comment]
> @zihaomu Anything else is blocking this PR from merge?
> 
> Update: we need to merge with benchmark results. Could you run benchmarking with this model? @zihaomu

This PR should be closed after the OpenCV support new DB API, since new DB API support pp-ocr V3 and V2.

**Comment by fengyuentau** [comment]
> This PR should be closed

Didn't we agree on adding this model (pp-ocr) in place of db? Why are we closing this PR?

**Comment by zihaomu** [comment]
We can directly support ppocr-DB v2 and v3 at new API.

**Pull request closed without merging**
